### PR TITLE
Add Arb<Date> support

### PR DESCRIPTION
**Pull request by vladeekkk**
1) added Arb for java.util.Date
2) wrote test for it

### CODE REVIEW

File: kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/dates.kt [hunk 1]
```diff
@@ -247,3 +248,17 @@ fun Arb.Companion.zonedDateTime(
    localDateTime(minValue, maxValue),
    zoneId
 ) { time, zone -> time.atZone(zone) }
+
+/**
+ * Arberates a stream of random [Date]
+ */
+fun Arb.Companion.date(
+   minDate: String = "1970-01-01",
+   maxDate: String = "2050-12-31",
```
Comment (vladeekkk) [inline kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/dates.kt:258]: working with java.util.Date is inconvenient, is it okay that default params are Strings?

Comment (Kantis) [reply kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/dates.kt:258]: What do you think about provide functions to use either Date or String in the API. 

`Arb.date()` feels like a very prioritized API for a class that probably shouldnt be used a lot (`java.util.Date`). What do you think about calling it `Arb.javaDate()`?

Comment (vladeekkk) [reply kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/dates.kt:258]: Yep, renaming to javaDate seems better ✔️ 

Also added second `Arb.javaDate()` function

**Comment by vladeekkk** [comment]
Some checks failed because `LocalDate.ofInstant()` is only in Java 9 and later, so I used `ZonedDateTime.ofInstant()` instead in `DateTest.kt`.

**Comment by vladeekkk** [comment]
https://github.com/kotest/kotest/issues/3503

**Review by Kantis — APPROVED** [review]

**Comment by Kantis** [comment]
Thank you for the contribution @vladeekkk 🙏🏼

**Pull request merged**